### PR TITLE
get rid of cpp atomic raw field

### PR DESCRIPTION
**Pull request by bung87**
refers to https://github.com/nim-lang/Nim/issues/20081#issuecomment-1328350093

this doesn't solve #20081 problem

### CODE REVIEW

File: tests/cpp/tatomic_sizeof.nim [hunk 1]
```diff
@@ -0,0 +1,8 @@
+discard """
+  targets: "cpp"
+"""
+
+import std/atomics
+
+doAssert sizeOf(Atomic[int]) == sizeOf(int)
+doAssert sizeOf(Atomic[bool]) == sizeOf(bool)
```
Comment (Araq) [inline tests/cpp/tatomic_sizeof.nim]: But these tests were green before too. I think.

Comment (bung87) [reply tests/cpp/tatomic_sizeof.nim]: there's `tests/stdlib/concurrency/tatomics_size.nim` , similar to this, so I can delete this test file I think.

File: compiler/sizealignoffsetimpl.nim [hunk 2]
```diff
@@ -408,8 +408,25 @@ proc computeSizeAlign(conf: ConfigRef; typ: PType) =
...
       typ.paddingAtEnd = typ.lastSon.paddingAtEnd
-
-  of tyGenericInst, tyDistinct, tyGenericBody, tyAlias, tySink, tyOwned:
+  
+  of tyGenericInst:
+    var index = typ.len - 1
+    if sfSizeOf in typ[0].sym.flags:
+      var id:int
+      for n in typ[0].sym.ast[0][1].sons:
+        if n.kind == nkCall and n[0].kind == nkIdent and n[0].ident.s == "sizeOf":
```
Comment (juancarlospaco) [inline compiler/sizealignoffsetimpl.nim:417]: This works with `sizeof()` ?. 🤔

Comment (bung87) [reply compiler/sizealignoffsetimpl.nim:417]: this feature added by this PR.

Comment (juancarlospaco) [reply compiler/sizealignoffsetimpl.nim:417]: I mean if code uses `sizeof()` instead of `sizeOf()`.

Comment (bung87) [reply compiler/sizealignoffsetimpl.nim:417]: hmm, didn't aware of that,  `system.sizeof` is all lower case ,pragmas are all camel case. so it's no.

Comment (juancarlospaco) [reply compiler/sizealignoffsetimpl.nim:417]: Should it ?. 🤔 Nim is case insensitive.

Comment (bung87) [reply compiler/sizealignoffsetimpl.nim:417]: dunno the answer, there're lot of places in compiler using string comparison

**Comment by Araq** [comment]
While I appreciate the effort and can see where you're coming from, this is a bad idea and we need something better for `Atomic[T]`. You cannot even construct this thing properly currently and removing the field while enabling `sizeof` solves very little, if anything.

**Comment by bung87** [comment]
why is that "You cannot even construct this thing properly " ?

**Comment by Araq** [comment]
```var x = Atomic[int](raw: 4)``` ?

**Comment by bung87** [comment]
but this PR is for removing raw field

**Comment by bung87** [comment]
other issue related to this is when atomic as an object field, object default constructor will use copy constructor while it's explicitly removed in cpp, wether use byRef pragma or wrapped as `std::atomic_ref` which is C++20 feature.

**Comment by jmgomez** [comment]
I may have an idea that can get around all these issues with the constructors in cpp but need to get the extended functionality for the constructor pragma first, and before that need to fix an existing issue with it.

But long story short it will be
```nim 
  type Foo = object 
    a : int 
    b : int = 10
```
    
    
That could generate a constructor in the cpp side of things with `Foo(NI b)` signature. 

So when instantiating an object it's automatically called (as it's marked with a new type flag that will carry it over). But it may also work to importcpp types too as it will call the already defined constructor. 

Not sure if that will work for all cases though.   

@Araq any thoughts? 

Edit: Wrong PR. But the question still holds though

**Comment by bung87** [comment]
still the atomic type size should be calculated

**Comment by jmgomez** [comment]
Yeah, I know. Sorry for the confusion

**Pull request closed without merging**